### PR TITLE
lambdify: Wrap input arguments with asarray when using numpy.

### PR DESCRIPTION
**Pull request by richardotis**
Fixes gh-11306.

I'm having issues with local test failures on sympy master (i.e., before this PR), so don't merge until tests pass (if they do).

CC @asmeurer

### CODE REVIEW

**Comment by asmeurer** [comment]
Looks good.

**Comment by richardotis** [comment]
Looks like the check for argument wrapping is too broad, and it's being incorrectly applied in some places.

**Comment by richardotis** [comment]
@asmeurer This is ready to merge if you're happy with the proposed solution and test.

**Comment by asmeurer** [comment]
I am. Hopefully this doesn't break anyone's code. It should solve more problems than it creates.

File: sympy/utilities/lambdify.py [hunk 1]
```diff
@@ -386,6 +386,14 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
     namespace.update({'builtins':builtins, 'range':range})
 
     func = eval(lstr, namespace)
+    # For numpy lambdify, wrap all input arguments in arrays.
+    # This is a fix for gh-11306.
+    if module_provided and _module_present('numpy',namespaces):
```
Comment (thisch) [inline sympy/utilities/lambdify.py]: a whitespace is missing before namespaces. Don't we run pep8 on all modules to catch such coding style violations?

Comment (richardotis) [inline sympy/utilities/lambdify.py]: I wrote it that way to be consistent with the appearance of the same function call earlier in the module. I do not know the SymPy policy on coding style.

Comment (asmeurer) [inline sympy/utilities/lambdify.py]: No, we don't enforce PEP 8 (only trailing whitespace, and some other issues), but we do generally try to follow it, so there should be a space after commas.

**Comment by moorepants** [comment]
Seeing a performance regression from this PR, see #14671.

**Pull request merged**